src/decode-vxlan.c

```diff
@@ -153,37 +165,60 @@ int DecodeVXLAN(ThreadVars *tv, DecodeThreadVars *dtv, Packet *p,
     uint16_t eth_type = SCNtohs(ethh->eth_type);
     SCLogDebug("VXLAN ethertype 0x%04x", eth_type);
 
-    switch (eth_type) {
-        case ETHERNET_TYPE_ARP:
-            SCLogDebug("VXLAN found ARP");
-            break;
-        case ETHERNET_TYPE_IP:
-            SCLogDebug("VXLAN found IPv4");
-            decode_tunnel_proto = DECODE_TUNNEL_IPV4;
-            break;
-        case ETHERNET_TYPE_IPV6:
-            SCLogDebug("VXLAN found IPv6");
-            decode_tunnel_proto = DECODE_TUNNEL_IPV6;
-            break;
-        case ETHERNET_TYPE_VLAN:
-        case ETHERNET_TYPE_8021AD:
-        case ETHERNET_TYPE_8021QINQ:
-            SCLogDebug("VXLAN found VLAN");
-            decode_tunnel_proto = DECODE_TUNNEL_VLAN;
-            break;
-        default:
-            SCLogDebug("VXLAN found unsupported Ethertype - expected IPv4, IPv6, VLAN, or ARP");
-            ENGINE_SET_INVALID_EVENT(p, VXLAN_UNKNOWN_PAYLOAD_TYPE);
+    /* TODO check further that we strip only once per real packet */
```

Comment:
concern here is that we may strip things that we don't want to strip, leading to a bypass